lib/common/CommonBasic.h

```diff
@@ -10,6 +10,20 @@
 
 #include "CommonPal.h"
 
+#if defined(__forceinline)
+// Make clang / g++ happy
+#    define CH_FORCE_INLINE __attribute__((always_inline))
+#else
+#    define CH_FORCE_INLINE __forceinline
+#endif
+
+#if defined(_MSC_VER)
+// MSVC template name lookup workarounds
+#    define CH_MEMBER_TYPENAME
+#else
+#    define CH_MEMBER_TYPENAME typename
+#endif
```

Comment:
* Please move this section to ```CommonPal.h``` (where we defined ```ABSTRACT```).
* For ```__forceinline```: On your clang/g++ system, was the definition coming from ```pal_mstypes.h```? If so, consider not introducing a new macro. ```#undef``` previous macro and ```#define __forceinline __attribute__((always_inline))```.
* For ```CH_MEMBER_TYPENAME```: Could you please just name it ```TYPENAME```, for the sake of being shorter and easier to read? We can deal with conflicts if ever running into one. (Discussed with other people, no ideal solution, generally prefer the shorter name.)